### PR TITLE
add --extra-gen-snapshot-options support for build_aar command.

### PR DESCRIPTION
**Pull request by 00SunnyDay00**
Tiny patch for https://github.com/flutter/flutter/pull/69592

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by 00SunnyDay00** [comment]
@googlebot I signed it!

**Review by jmagman — CHANGES_REQUESTED** [review]
This needs tests.  See existing `build_apk_test` for example.
https://github.com/flutter/flutter/blob/9e88fe328efcf42d3037a965b39269f0805cd4ce/packages/flutter_tools/test/commands.shard/permeable/build_apk_test.dart#L213

**Comment by 00SunnyDay00** [comment]
> This needs tests. See existing `build_apk_test` for example.
> 
> https://github.com/flutter/flutter/blob/9e88fe328efcf42d3037a965b39269f0805cd4ce/packages/flutter_tools/test/commands.shard/permeable/build_apk_test.dart#L213



> This needs tests. See existing `build_apk_test` for example.
> 
> https://github.com/flutter/flutter/blob/9e88fe328efcf42d3037a965b39269f0805cd4ce/packages/flutter_tools/test/commands.shard/permeable/build_apk_test.dart#L213

done

**Comment by 00SunnyDay00** [comment]
done

File: packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart [hunk 1]
```diff
@@ -219,6 +224,28 @@ void main() {
...
+
+    testUsingContext('support ExtraDartFlagOptions',
+            () async {
+      final String projectPath = await createProject(tempDir,
+          arguments: <String>['--no-pub', '--template=module']);
+
+      globals.fs.directory(globals.fs.path.join(tempDir.path, 'flutter_project',
+          'build', 'host','outputs','repo')).createSync(recursive: true);
+
+      await runBuildAarCommand(projectPath,
```
Comment (jmagman) [inline packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart:236]: This isn't testing that the options are used.  Take another look at how `build_apk` is validating the process being called. https://github.com/flutter/flutter/blob/9e88fe328efcf42d3037a965b39269f0805cd4ce/packages/flutter_tools/test/commands.shard/permeable/build_apk_test.dart#L213

Comment (00SunnyDay00) [reply packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart]: done

Comment (00SunnyDay00) [reply packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart]: done

File: packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart [hunk 2]
```diff
@@ -219,6 +232,47 @@ void main() {
...
+      ]), throwsToolExit(message: 'Gradle task assembleAarRelease failed with exit code 1'));
+      expect(processManager, hasNoRemainingExpectations);
+    },
+        overrides: <Type, Generator>{
+          AndroidSdk: () => mockAndroidSdk,
+          FlutterProjectFactory: () => FakeFlutterProjectFactory(tempDir),
+          ProcessManager: () => processManager,
+          FeatureFlags: () => TestFeatureFlags(isIOSEnabled: false),
+          AndroidStudio: () => FakeAndroidStudio(),
+        });
```
Comment (jmagman) [inline packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart:275]:
```suggestion
    },
    overrides: <Type, Generator>{
      AndroidSdk: () => mockAndroidSdk,
      FlutterProjectFactory: () => FakeFlutterProjectFactory(tempDir),
      ProcessManager: () => processManager,
      FeatureFlags: () => TestFeatureFlags(isIOSEnabled: false),
      AndroidStudio: () => FakeAndroidStudio(),
    });
```

File: packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart [hunk 3]
```diff
@@ -219,6 +232,47 @@ void main() {
...
+          '-Pis-plugin=false',
+          '-PbuildNumber=1.0',
+          '-q',
+          '-Ptarget=${globals.fs.path.join('lib', 'main.dart')}',
+          '-Pdart-obfuscation=false',
+          '-Pextra-front-end-options=foo,bar',
+          '-Ptrack-widget-creation=true',
+          '-Ptree-shake-icons=true',
+          '-Ptarget-platform=android-arm,android-arm64,android-x64',
+          'assembleAarRelease'
```
Comment (jmagman) [inline packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart:256]:
```suggestion
          'assembleAarRelease',
```

File: packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart [hunk 4]
```diff
@@ -259,3 +313,15 @@ class FakeAndroidBuilder extends Fake implements AndroidBuilder {
     this.buildNumber = buildNumber;
   }
 }
+
+class FakeAndroidSdk extends Fake implements AndroidSdk {
+  FakeAndroidSdk(this.directory);
+
+  @override
+  final Directory directory;
+}
```
Comment (jmagman) [inline packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart:322]: Overriding `directory` isn't necessary.
```suggestion
class FakeAndroidSdk extends Fake implements AndroidSdk { }
```

File: packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart [hunk 5]
```diff
@@ -190,9 +194,18 @@ void main() {
...
   group('Gradle', () {
     Directory tempDir;
+    AndroidSdk mockAndroidSdk;
+    String gradlew;
+    FakeProcessManager processManager;
+    String flutterRoot;
 
     setUp(() {
       tempDir = globals.fs.systemTempDirectory.createTempSync('flutter_tools_packages_test.');
+      mockAndroidSdk = FakeAndroidSdk(globals.fs.directory('irrelevant'));
```
Comment (jmagman) [inline packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart:204]: Overriding `directory` isn't necessary.
```suggestion
      mockAndroidSdk = FakeAndroidSdk();
```

**Review by jmagman — COMMENTED** [review]
Some nits.
@blasten anything to add?

File: packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart (hunk 2 above)
Comment (00SunnyDay00) [reply packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart]: done

File: packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart (hunk 3 above)
Comment (00SunnyDay00) [reply packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart]: done

File: packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart (hunk 4 above)
Comment (00SunnyDay00) [reply packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart]: done

File: packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart (hunk 5 above)
Comment (00SunnyDay00) [reply packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart]: done

**Review by jmagman — APPROVED** [review]
LGTM awaiting review from @blasten to merge.

**Review by blasten — APPROVED** [review]
LGTM

**Pull request merged**